packages/ua-devtools-evm-hardhat/src/oapp/typescript/typescript.ts

```diff
@@ -0,0 +1,427 @@
+            factory.createIdentifier(OPTIONAL_DVN_THRESHOLD),
+            factory.createNumericLiteral(optionalDVNThreshold)
+        ),
+    ])
+}
+
+/**
+ * Creates a send configuration object.
+ *
+ * @param {Uln302ExecutorConfig} sendDefaultExecutorConfig - The default executor configuration.
+ * @param {Uln302UlnConfig} sendDefaultUlnConfig - The default ULN configuration.
+ * @returns {PropertyAssignment} A PropertyAssignment object representing the send configuration.
+ *
+ *     sendConfig: {
+ *       executorConfig: {},
+ *       ulnConfig: {},
+ *     }
+ */
+export const creatSendConfig = (
+    sendDefaultExecutorConfig: Uln302ExecutorConfig,
+    sendDefaultUlnConfig: Uln302UlnConfig
+): PropertyAssignment => {
+    return factory.createPropertyAssignment(
+        factory.createIdentifier(SEND_CONFIG),
+        factory.createObjectLiteralExpression([
+            factory.createPropertyAssignment(
+                factory.createIdentifier(EXECUTOR_CONFIG),
+                creatExecutorConfig(sendDefaultExecutorConfig)
+            ),
+            factory.createPropertyAssignment(
+                factory.createIdentifier(ULN_CONFIG),
+                creatUlnConfig(sendDefaultUlnConfig)
+            ),
+        ])
+    )
+}
+
+/**
+ * Creates a receive configuration object.
+ *
+ * @param {Uln302UlnConfig} receiveDefaultUlnConfig - The default ULN configuration.
+ * @returns {PropertyAssignment} A PropertyAssignment object representing the receive configuration.
+ *
+ *     receiveConfig: {
+ *       ulnConfig: {}
+ *     }
+ */
+export const creatReceiveConfig = (receiveDefaultUlnConfig: Uln302UlnConfig): PropertyAssignment => {
+    return factory.createPropertyAssignment(
+        factory.createIdentifier(RECEIVE_CONFIG),
+        factory.createObjectLiteralExpression([
+            factory.createPropertyAssignment(
+                factory.createIdentifier(ULN_CONFIG),
+                creatUlnConfig(receiveDefaultUlnConfig)
+            ),
+        ])
+    )
+}
+
+/**
+ * Creates a default LayerZero configuration from the passed in networks by reading current defaults off-chain.
+ *
+ * @param {string} fromNetwork - The source network.
+ * @param {string} toNetwork - The destination network.
+ * @returns {Promise<PropertyAssignment>} A promise that resolves to a PropertyAssignment object representing the default configuration.
+ * @throws {Error} Throws an error if any required default configuration is missing.
+ *
+ *         config: {
+ *              sendLibrary: "0x0000000000000000000000000000000000000000",
+ *              receiveLibraryConfig: {...},
+ *              receiveLibraryTimeoutConfig: {...},
+ *              sendConfig: {...},
+ *              receiveConfig: {...}
+ *         }
+ */
+export const createDefaultConfig = async (fromNetwork: string, toNetwork: string): Promise<PropertyAssignment> => {
+    const receiveDefaultConfig = await getReceiveConfig(fromNetwork, toNetwork)
```

Comment:
Could do these in parallel.  I.e.,

```
const [receiveDefaultConfig, sendDefaultConfig] = await Promise.all([getReceiveConfig(fromNetwork, toNetwork), getSendConfig(fromNetwork, toNetwork)])
```

Further, you should probably use `backOff(...)` as even strong RPCs tend to have some read errors:


```
const [receiveDefaultConfig, sendDefaultConfig] = await Promise.all([backOff(()=>getReceiveConfig(fromNetwork, toNetwork)), backOff(()=>getSendConfig(fromNetwork, toNetwork)]))
```

My concern is a single RPC error may really mess up the entire task run.